data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -927,6 +903,12 @@ impl TraceExporterBuilder {
         self
     }
 
+    /// Set the session token
+    pub fn set_test_session_token(&mut self, test_session_token: &str) -> &mut Self {
```

Comment:
What happens if you set the test session token and you're not in a test env, or if you're talking to a real agent? Shouldn't we return an error or warning? We know the env at build time so we can check that it's not prod. The info endpoint says "test" in the version of the test agent, so we can know when we are sending traces.

The doc comment could be more thorough as well and explain when the token can be used.

Reply:
Even if nothing breaks, it's a potential point of confusion for a user of the API. If the token is being set when it can't be used it's either a) an oversight by the API user or b) they think they're going to get behavior that isn't going to work. I'd be less concerned if this was behind a feature, but this is going to be exposed everywhere. 

I'd prefer an error, but I'm ok with better documentation and a warning since it won't break production workflows.

Reply:
@ekump By the point in time you set this, the info endpoint has not been queried yet. So you'd only see this when fetching the info endpoint with the test session token being set. If you really insist that one could do it, but I don't think it's important enough.

Reply:
@bwoebi To unblock this PR I'm ok with just adding better documentation. Checking `DD_ENV` isn't prod and issuing a warning would be nice to have. 

At some point the `TraceExporterBuilder` is probably going to have to hit `/info` because there are certain features that are only available in newer trace agents. But we can discuss that separately.